### PR TITLE
Cocktail show page

### PR DESCRIPTION
**Pull request by zachjamesgreen**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Build is passing
## Pull request type
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):
## What is the new behavior?

- user can see cocktail details on recommendation page
- user can see cocktail details on details page after search
- Tagline can be changed depending on which page the user is on
## Does this introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/cocktail.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  render() {
+    let { cocktail: {id, attributes}, isLoaded } = this.state
+    let { name, recipe, instructions, thumbnail, rating } = attributes
+    instructions = instructions.split(/(?<=\.)/)
+    return (
+      <section className="flex justify-evenly">
+        <div>
+          <p className="font-playfair italic text-3xl pb-3">{this.props.tagline}</p>
+          <h1 className="font-playfair font-black text-6xl pb-2">{ name }</h1>
+          <p className="font-montserrat italic">Rate me to get better recommendations</p>
```
Comment (mkrumholz) [inline src/components/cocktail.js:25]: Classy

File: src/components/cocktail.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                  return <li key={i}>{ingredient}</li>
+                })}
+              </ul>
+            </div>
+            <div className="max-w-xs">
+              <p className="font-montserrat font-bold uppercase">Directions</p>
+              <ol className="list-decimal ml-4">
+                {instructions.map((step, i) => {
+                  return <li key={i}>{step}</li>
+                })}
```
Comment (mkrumholz) [inline src/components/cocktail.js:43]: Nice! Way to get these to display in steps.

File: src/components/show_page.js [hunk 3]
```diff
@@ -1,59 +1,43 @@
...
-        </div>
-
-        <SocialLinks cnames="mx-4 space-y-8" />
-      </div>
-    </section>
-  )
-}
+  componentDidMount() {
+    if (this.id) {
+      let url = `http://localhost:3001/api/v1/cocktails/${this.id}`
```
Comment (mkrumholz) [inline src/components/show_page.js:19]: Are we going to need to update these URLs for production?

File: src/test/cocktail.test.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+it("renders for search details", async () => {
+  render(<Cocktail cocktail={data} tagline={'Have a...'} />)
+  const text = screen.getByText(/Have a.../)
+  expect(text).toBeInTheDocument()
+  const ingredients = screen.getByText(/Ingredients/)
+  expect(ingredients).toBeInTheDocument()
+  const directions = screen.getByText(/Directions/)
+  expect(directions).toBeInTheDocument()
+});
```
Comment (mkrumholz) [inline src/test/cocktail.test.js:41]: Cool to see the tests in here!

File: src/test/show_page.test.js [hunk 5]
```diff
@@ -18,19 +18,6 @@ import { getByTestId, render, screen } from '@testing-library/react';
 //   container = null;
 // });
```
Comment (mkrumholz) [inline src/test/show_page.test.js:20]: Are these left in intentionally?

**Review by mkrumholz — APPROVED** [review]
I'm going to approve, since I can't honestly say I can follow this all. But definitely check the links and comments to make sure those are as they should be!

File: src/components/cocktail.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    this.state = {
+      isLoaded: false,
+      cocktail: this.props.cocktail
+    }
+  }
+
+
+  render() {
+    let { cocktail: {id, attributes}, isLoaded } = this.state
+    let { name, recipe, instructions, thumbnail, rating } = attributes
```
Comment (markcyen) [inline src/components/cocktail.js:18]: I think we have more attributes, so you can add glass if you'd like.

File: src/components/cocktail.js (hunk 1 above)
Comment (markcyen) [reply src/components/cocktail.js:25]: Very... 🍸

File: src/test/cocktail.test.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import Cocktail from '../components/cocktail.js';
+import { getByTestId, render, screen, waitFor } from '@testing-library/react';
+
+const data = {
+  id: "11007",
+  type: "cocktail",
+  attributes: {
+    name: "Margarita",
```
Comment (markcyen) [inline src/test/cocktail.test.js:9]: You couldn't pick the Godfather as a test example... thanks Zach 🥃

**Review by markcyen — APPROVED** [review]
Looks sweet... like 🍭

**Pull request merged**